### DISCUSSION TITLE
Directional spectrum

### DISCUSSION THREAD

**Discussion by sarriegi** [NOAA-EMC/WW3#874] [Q&A]
Hi, how can I get the directional spectrum as output from a WW3 run? Is it the 'efth' variable in the point output file? I did a Hurricane Ian run, choosing 95 points for spectral output. The efth variable had zero values for all points and all times. The significant wave height values are reasonable, so that makes me think that I am doing something wrong with my ww3_ounp file. I am attaching it here along with my ww3_multi file, in case someone can spot what causes the problem.

@aliabdolali 

Angelos

**Comment by aliabdolali** [comment]
Try type 1 subtype 3

**Reply by sarriegi** [reply]
Hi Ali, so should the line in ww3_ounp.inp file be like this:
![image](https://user-images.githubusercontent.com/77630268/210152050-a0a081b7-931a-4899-970f-340d19a98ad8.png)
Then I get this error:
![image](https://user-images.githubusercontent.com/77630268/210152070-8766a384-7cc2-4522-bc60-8f58aec3cc05.png)

Angelos

**Reply by aliabdolali** [reply]
You set it wrong,
You should define https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_ounp.inp#L33 to 1 which defines type and then uncomment https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_ounp.inp#L51 to 3 1 0 4 and comment the following lines. https://github.com/NOAA-EMC/WW3/blob/develop/model/inp/ww3_ounp.inp is what you want to generate directional spectral density.

**Reply by sarriegi** [reply]
Thank you Ali! I still get zero values for all the points and times for the efth variable. Could I have something wrong in my ww3_multi.inp file or my ww3_grid.inp file? (I am attaching the ww3_grid, ww3_multi and ww3_ounp files here, I realized I didn't attached them in my initial message).

Angelos
[ww3_multi.txt](https://github.com/NOAA-EMC/WW3/files/10327954/ww3_multi.txt)
[ww3_ounp.txt](https://github.com/NOAA-EMC/WW3/files/10327955/ww3_ounp.txt)
[ww3_grid.txt](https://github.com/NOAA-EMC/WW3/files/10327958/ww3_grid.txt)

**Reply by ukmo-ccbunney** [reply]
@sarriegi Have you checked you wind input? That is the only source of wave generation in your model. You should be able to see the windspeed and direction in the output from ww3_ounp. Make sure these are not zero.

**Reply by sarriegi** [reply]
Thank you @ukmo-ccbunney! Indeed the wind output from the ww3_ounp output is zero everywhere and the winddir output from ww3_ounp is 270 everywhere. However, the wind, significant wave height and other variables from the ww3_ounf output all make sense. Do you have any idea of what could lead to a correct ww3_ounf output but wrong ww3_ounp output?

Angelos

**Reply by aliabdolali** [reply]
> Thank you @ukmo-ccbunney! Indeed the wind output from the ww3_ounp output is zero everywhere and the winddir output from ww3_ounp is 270 everywhere. However, the wind, significant wave height and other variables from the ww3_ounf output all make sense. Do you have any idea of what could lead to a correct ww3_ounf output but wrong ww3_ounp output?
> 
> Angelos

270 degrees is the default and where there is no input to the model. Since you are getting reasonable field outputs, could you try ww3_outp to see if there is sth wrong with NetCDF or if the problem persists in ASCII outputs?
Also, I would recommend trying a different version of WW3, to see if this is version-dependent or not.

**Reply by ukmo-ccbunney** [reply]
> Thank you @ukmo-ccbunney! Indeed the wind output from the ww3_ounp output is zero everywhere and the winddir output from ww3_ounp is 270 everywhere. However, the wind, significant wave height and other variables from the ww3_ounf output all make sense. Do you have any idea of what could lead to a correct ww3_ounf output but wrong ww3_ounp output?

Well that is very odd. I'm not very familiar with the unstructured grids - but are you sure that all your point output requests are on wet sea points? I.e. not on land or outside the domain? Check that the WW3_SHEL program is not reporting the points are not on sea points.

**Reply by sarriegi** [reply]
@aliabdolali @ukmo-ccbunney The ww3_outp output still had zero values everywhere and 270 values for the wind direction. These results are for the unstructured version of WW3 inside the CoastalApp. I got the same results by running the unstructured latest version of WW3. However, when I ran the WW3 version inside the CoastalApp with a structured grid, I got non-zero values for the wind and spectrum and non-270 values for the wind direction in the ww3_ounp output. Maybe this is a stupid question but does the ww3_ounp file require any specific flags in the switch file when it is compiled for an unstructured grid?

Angelos

**Reply by aliabdolali** [reply]
The point output does not need a specific switch. I am puzzled as we do have a couple of regtests for unstr meshes and point outputs i.e. (ww3_tp2.21, ww3_tp2.17, ...), could you try one of them?
Also, as  Chris mentioned, if the points are located on the land, the model does not generate point outputs for them, and you can see the list of wet points in the log.ww3 and ww3_shel.out.
One other thing, could you try ww3_shel instead of ww3_multi? We use ww3_multi in the coastal_app but they should behave the same for unstr.

**Reply by sarriegi** [reply]
@aliabdolali @ukmo-ccbunney I forgot to mention that all the points are in the open ocean. I tried the ww3_tp2.21 and it works fine for both ww3_shel and ww3_multi. Then I tried again my case of Hurricane Ian with unstructured grid and the ww3_shel instead of ww3_multi and it also worked fine. For some reason this case gives zero values for the wind and 270 values for the wind direction when I use the ww3_multi file and unstructured grid.

Angelos

**Reply by aliabdolali** [reply]
that is great info, I think the problem is with ww3_multi.inp and your mod_def.points grid. Could you add your ww3_grid.inp here?

**Reply by sarriegi** [reply]
@aliabdolali Yes, I am attaching it here.

Angelos
[ww3_grid.txt](https://github.com/NOAA-EMC/WW3/files/10354229/ww3_grid.txt)

**Reply by aliabdolali** [reply]
oops, I meant ww3_multi.inp

**Reply by sarriegi** [reply]
@aliabdolali Here it is.

[ww3_multi.txt](https://github.com/NOAA-EMC/WW3/files/10354310/ww3_multi.txt)

**Reply by aliabdolali** [reply]
could you try the attached one? 
[ww3_multiAA.txt](https://github.com/NOAA-EMC/WW3/files/10354322/ww3_multiAA.txt)

**Reply by sarriegi** [reply]
@aliabdolali It worked! Thank you very much! So my mistake was that the line saying 'points' was not commented out?
